docs-#5963: add custom query example to docs (#5964)
vuestorefront/vue-storefront: packages/core/docs/advanced/extending-graphql-queries.md

```ts
const { search } = useProduct();

search({
  customQuery: {
    products: 'my-products-query',
    metadata: { size: 'xl' }
  }
}); 
```

In the example above, we are changing `products` query, and our function that will take care of this overriding is `my-products-query`. Additionally, we used `metadata` field to send information about the product we seek for. As a second step, we need to define that function.

Each custom query lives in the `middleware.config.js`, so it's the place where we should define `my-products-query`.

Customizing variables:

```js
module.exports = {
  integrations: {
    ct: {
      location: '@vue-storefront/commercetools-api/server',
      configuration: { /* ... */ },
      customQueries: {
        'my-products-query': ({ query, variables, metadata }) => {

          variables.locale = 'en'
          variables.size = metadata.size

          return { query, variables }
        }
      }
    }
  }
};
```

Writing custom GraphQL queries:

```js

> module.exports = {

Reviewer: ```suggestion
// middleware.config.js

module.exports = {
```

  integrations: {
    ct: {
      location: '@vue-storefront/commercetools-api/server',
      configuration: { /* ... */ },
      customQueries: {
        'my-products-query': ({ variables }) => {
          query: `
            query products($where: String) {
              products(where: $where) { /* ... */ }
            }
          `,
          variables,
        }
      }
    }
  }
};
```

> In case of many custom queries extending the default ones, we recommend extracting them into separate folder to not overload `middleware.config.js` with GraphQL queries:

Reviewer: ```suggestion
## Keeping the configuration tidy

Your configuration file can quickly become a mess if you override a lot of queries. We recommend extracting them into a separate folder to not overload `middleware.config.js` with GraphQL queries.

You can create a new file (or files) that exports the queries. Then, you can import them in the `middleware.config.js`:
```

```js
// customQueries/index.js

module.exports = {
  'my-products-query': ({ variables }) => {
    query: `
      query products($where: String) {
        products(where: $where) { /* ... */ }
      }
    `,
    variables,
  },
  ... // other custom queries
};
```

And then import it in `middleware.config.js`:

> ```js
> const customQueries = require('./customQueries');
> 
> module.exports = {

Reviewer: ```suggestion
```js
// middleware.config.js

const customQueries = require('./customQueries');

module.exports = {
```
